ui/app/components/download-button.js

```diff
@@ -12,17 +12,37 @@ export default Ember.Component.extend({
     return `${this.get('filename')}-${new Date().toISOString()}.${this.get('extension')}`;
   }),
 
-  href: computed('data', 'mime', 'stringify', function() {
+  fileLike: computed('data', 'mime', 'strigify', 'download', function() {
+    let file;
     let data = this.get('data');
-    const mime = this.get('mime');
+    let filename = this.get('download');
+    let mime = this.get('mime');
     if (this.get('stringify')) {
       data = JSON.stringify(data, null, 2);
     }
+    if (window.navigator.msSaveOrOpenBlob) {
+      file = new Blob([data], { type: mime });
+      file.name = filename;
+    } else {
+      file = new File([data], filename, { type: mime });
+    }
+    return file;
+  }),
 
-    const file = new File([data], { type: mime });
-    return window.URL.createObjectURL(file);
+  href: computed('fileLike', function() {
+    return window.URL.createObjectURL(this.get('fileLike'));
   }),
 
+  click(event) {
+    if (!window.navigator.msSaveOrOpenBlob) {
+      return;
+    }
+    event.preventDefault();
+    let file = this.get('fileLike');
+    //lol whyyyy
+    window.navigator.msSaveOrOpenBlob(file, file.name);
```

Comment:
Wow, this is annoying. Makes it harder to pretend that browsers are all the same.

It would be nice to hide all this `msSaveOrOpenBlob` stuff in some polyfill rather than in this component, but if it's only going to be used here, I guess it's not so bad.

Reply:
Yeah @johncowen mentioned https://github.com/eligrey/FileSaver.js/ which may be worth looking at if this gives us any other issues. I think for now since it's _just_ in this component we'll try this for now.